### PR TITLE
all: let interop test use shaded dependency correctly

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

File: netty/shaded/build.gradle [hunk 1]
```diff
@@ -18,7 +18,6 @@ dependencies {
             project(':grpc-testing-proto'),
             project(':grpc-testing'),
             libraries.truth
-    shadow project(':grpc-core')
```
Comment (ejona86) [inline netty/shaded/build.gradle]: I figured out why this is here: it is necessary for the pom to include a dependency on grpc-core.

File: xds/build.gradle [hunk 2]
```diff
@@ -39,11 +38,16 @@ dependencies {
 
     testCompile project(':grpc-core').sourceSets.test.output
 
-    compileOnly libraries.javax_annotation
+    compileOnly libraries.javax_annotation,
+            // The unshaded netty_epoll should not be a transitive dependency.
```
Comment (ejona86) [inline xds/build.gradle]: This comment sort of just reexplains what a compileOnly dependency is. Instead how about:
```
      // At runtime use the epoll included in grpc-netty-shaded
```

Comment (dapengzhang0) [reply xds/build.gradle]: Done.

File: netty/shaded/build.gradle (hunk 1 above)
Comment (dapengzhang0) [reply netty/shaded/build.gradle]: I see. Changed back.

**Comment by dapengzhang0** [comment]
gae-interop-testing was failing https://source.cloud.google.com/results/invocations/24432bbe-0f09-4149-89dd-7190c766cf73/targets
because the test uses ManagedChannelBuilder and then NettyChannelBuilder
```java
assertTrue(builder instanceof NettyChannelBuilder);
((NettyChannelBuilder) builder).flowControlWindow(65 * 1024);
```

**Comment by ejona86** [comment]
This PR caused a regression in the ALTS tests. https://source.cloud.google.com/results/invocations/691d9965-fea1-487d-b606-352a5234039e/targets/grpc%2Fcore%2Fpull_request%2Flinux%2Fgrpc_interop_toprod/log

```
2020-03-01 20:02:12,491 Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/lang3/SystemUtils
	at io.grpc.alts.CheckGcpEnvironment.isRunningOnGcp(CheckGcpEnvironment.java:69)
	at io.grpc.alts.CheckGcpEnvironment.isOnGcp(CheckGcpEnvironment.java:44)
	at io.grpc.alts.ComputeEngineChannelBuilder.<init>(ComputeEngineChannelBuilder.java:62)
	at io.grpc.alts.ComputeEngineChannelBuilder.forTarget(ComputeEngineChannelBuilder.java:72)
	at io.grpc.alts.ComputeEngineChannelBuilder.forAddress(ComputeEngineChannelBuilder.java:77)
	at io.grpc.testing.integration.TestServiceClient$Tester.createChannel(TestServiceClient.java:399)
	at io.grpc.testing.integration.AbstractInteropTest.setUp(AbstractInteropTest.java:309)
	at io.grpc.testing.integration.TestServiceClient.setUp(TestServiceClient.java:198)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:56)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.SystemUtils
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 9 more
```

It can be trivially reproduced by running on your machine (no need for any other setup):
```
$ ./build/install/grpc-interop-testing/bin/test-client --use_alts=true
```

@dapengzhang0, we should probably revert this change while we try to figure out the solution.

**Comment by vjpai** [comment]
Cc: @jtattermusch

**Comment by dapengzhang0** [comment]
Trying to fix the issue of missing transitive dependencies for shadow configuration in #6792.

**Pull request merged**